motorApp/MclennanSrc/drvPM304.cc

```diff
@@ -719,6 +719,18 @@ STATIC int motor_init()
                         cntrl->use_encoder[motor_index] = 1;
                     }
                 }
+                /* Querying speeds for this axis */
+                sprintf(command, "%dQS", motor_index+1);
+                send_recv_mess(card_index, command ,buff);
+                /* splice creep speed - split up spaces then parse second integer as creep speed */
+                const char s[2] = " ";
+                char *token;
+                token = strtok(buff, s);
+                for (int i=0;i<2;i++) {
+                    token = strtok(NULL, s);
+                }
+                int creep_speed = atoi(token);
+                cntrl->creep_speeds[motor_index] = creep_speed;
 
```

Comment:
I am concerned this is not general enough. The P600 manual says it returns e.g. `01:SC = 700 SV = 16200 SA = 50000 SD = 100000 LD = 200000` but the PM304 either doesn't support the command or returns `SV=16200,SC=1000,SA=100000,SD=100000` depending on where I look.

Reply:
Ok, needs some work. In the PM304 manual there is a typo, but `xQS` should always return `SV=16200,SC=1000,SA=100000,SD=100000` so will modify to support that

Reply:
have updated. Luckily if you use `=` as the token instead of ` ` you get the same result as the second iteration gets the number.

Reply:
I've just pushed some changes to parse the line for SC in case the format were to be different and we ended up setting a silly value.

Reply:
Cool, looks good, I wonder if we should pull `"%dSC%d;"` out into a const as it's called 3 times

Reply:
Actually I think that's a rubbish idea - ignore that comment!